unit_testing/test_reports.py

```diff
@@ -3,38 +3,46 @@
 # pytest unit tests for spinalcordtoolbox.reports
 
 import pytest
+from pytest_cases import parametrize_with_cases
 import numpy as np
 
 from spinalcordtoolbox.image import Image
 from spinalcordtoolbox.reports.slice import Sagittal
+from spinalcordtoolbox.utils.sys import sct_test_path
 
 
-def labeled_data_test_params(path_in='sct_testing_data/t2/t2.nii.gz',
-                             path_seg='sct_testing_data/t2/labels.nii.gz'):
-    """Generate image/label pairs for various test cases of
-    test_sagittal_slice_get_center_spit."""
-    im_in = Image(path_in)            # Base anatomical image
-    im_seg_labeled = Image(path_seg)  # Base labeled segmentation
-    assert np.count_nonzero(im_seg_labeled.data) >= 2, "Labeled segmentation image has fewer than 2 labels"
+def im_base(path_in=sct_test_path('t2', 't2.nii.gz')):
+    # Base anatomical image
+    return Image(path_in)
 
+
+def im_seg_labeled(path_seg=sct_test_path('t2', 'labels.nii.gz')):
+    # Base labeled segmentation
+    im_seg = Image(path_seg)
+    assert np.count_nonzero(im_seg.data) >= 2, "Labeled segmentation image has fewer than 2 labels"
+    return im_seg
+
+
+def im_seg_one_label(im_seg=im_seg_labeled()):
     # Create image with all but one label removed
-    im_seg_one_label = im_seg_labeled.copy()
+    im_seg_one_label = im_seg.copy()
     for x, y, z in np.argwhere(im_seg_one_label.data)[1:]:
         im_seg_one_label.data[x, y, z] = 0
+    return im_seg_one_label
 
+
+def im_seg_no_labels(im_seg=im_seg_labeled()):
     # Create image with no labels
-    im_seg_no_labels = im_seg_labeled.copy()
+    im_seg_no_labels = im_seg.copy()
     for x, y, z in np.argwhere(im_seg_no_labels.data):
         im_seg_no_labels.data[x, y, z] = 0
-
-    return [pytest.param(im_in, im_seg_labeled, id='multiple_labels'),
-            pytest.param(im_in, im_seg_one_label, id='one_label'),
-            pytest.param(im_in, im_seg_no_labels, id='no_labels')]
+    return im_seg_no_labels
 
 
-@pytest.mark.parametrize('im_in,im_seg', labeled_data_test_params())
+@parametrize_with_cases("im_in", cases=[im_base])
+@parametrize_with_cases("im_seg", cases=[im_seg_labeled, im_seg_one_label, im_seg_no_labels])
 def test_sagittal_slice_get_center_spit(im_in, im_seg):
-    """Test that get_center_split returns a valid index list."""
+    """Test that get_center_spit returns a valid index list."""
```

Comment:
?? typo?

I think this is probably supposed to be "center_split".

Reply:
Oh, right, I forgot about this.

The function is called `get_center_spit`, but I never really questioned _why_ it was called that. Maybe it... IS a typo.

There's also another typo'd function I saw that I've been meaning to fix....

https://github.com/neuropoly/spinalcordtoolbox/blob/639ca81438763346b9fd4271c3fb696e6febd44a/spinalcordtoolbox/scripts/sct_analyze_lesion.py#L103